src/fdcache_entity.cpp

```diff
@@ -643,8 +644,8 @@ int FdEntity::Open(const headers_t* pmeta, off_t size, const struct timespec& ts
                 S3FS_PRN_WARN("failed to save cache stat file(%s), but continue...", path.c_str());
             }
         }
-        if(pcfstat){
-            delete pcfstat;
+        if(pcfstat.get() != nullptr){
+            delete pcfstat.release();
```

Comment:
This seems to be an unusual way to use `unique_ptr`, instead lifetimes should automatically delete this.

Reply:
Removed.